Replace spirv.hpp with spirv.hpp11 in objdump

jan-kelemen opened the pull request:
Fixes `-Wodr` warning with GCC, objdump seems to be the only remaining place where `spirv.hpp` header is used.
```
/mnt/work/git/SPIRV-Tools/external/spirv-headers/include/spirv/unified1/spirv.hpp:1483:6: warning: type ‘spv::Op’ violates the C++ One Definition Rule [-Wodr]
 1483 | enum Op {
      |      ^
/mnt/work/git/SPIRV-Tools/external/spirv-headers/include/spirv/unified1/spirv.hpp11:1483:12: note: an enum with different value name is defined in another translation unit
 1483 | enum class Op : unsigned {
      |            ^
/mnt/work/git/SPIRV-Tools/external/spirv-headers/include/spirv/unified1/spirv.hpp:2254:5: note: name ‘OpMax’ differs from name ‘Max’ defined in another translation unit
 2254 |     OpMax = 0x7fffffff,
      |     ^
/mnt/work/git/SPIRV-Tools/external/spirv-headers/include/spirv/unified1/spirv.hpp11:2254:5: note: mismatching definition
 2254 |     Max = 0x7fffffff,
      |     ^
```

dneto0 commented on tools/objdump/extract_source.cpp:
```diff
@@ -66,7 +66,7 @@ spv_result_t extractOpString(const spv_position_t& loc,
                              std::string* output) {
   assert(output != nullptr);
-  assert(instruction.opcode == spv::Op::OpString);
+  assert(instruction.opcode == (unsigned)spv::Op::OpString);
```
Generally we try to follow the advice in the Google style guide.

It advises against using C-style casts. See
See https://google.github.io/styleguide/cppguide.html#Casting 

I'm not bothered by these instances.

dneto0 reviewed (approved):
Thanks!

jan-kelemen commented:
The kokoro run was broken, but I think this had nothing to do with the source code changes.

Fixed the casts to be according to the style guide and rebased to main.

jan-kelemen commented:
Hopefully it's correct now, `CI-check-format` had some complaints.

jan-kelemen commented:
> Found formatting errors!
--- source/ext_inst.cpp	(before formatting)
+++ source/ext_inst.cpp	(after formatting)

Let me know what I should do, the `CI-check-format` is failing but due to a file which isn't the part of the PR.

Thanks!

s-perron commented:
Try rebasing on the latest main to fix the format error.
